pkg/ring/kv/memberlist/memberlist_client.go

```diff
@@ -288,11 +298,11 @@ func NewKV(cfg KVConfig) (*KV, error) {
 			return nil, err
 		}
 
-		if err != nil {
-			level.Error(util.Logger).Log("msg", "failed to join memberlist cluster", "err", err)
-		} else {
-			level.Info(util.Logger).Log("msg", "joined memberlist cluster", "reached_nodes", reached)
+		if reached == 0 && cfg.MaxJoinRetries > 0 {
+			go mlkv.retryJoinWithBackoff(cfg.JoinMembers)
 		}
+
+		level.Info(util.Logger).Log("msg", "joined memberlist cluster", "reached_nodes", reached)
```

Comment:
I think this should be in `else` branch, otherwise it's not correct (It will log "joined memberlist cluster, reached_nodes=0").